--- a/src/extension.js
+++ b/src/extension.js
@@ -6,7 +6,6 @@ import Gio from 'gi://Gio';
 import GLib from 'gi://GLib';
 import GObject from 'gi://GObject';
 import St from 'gi://St';
-import Cogl from 'gi://Cogl';
 import {Extension} from 'resource:///org/gnome/shell/extensions/extension.js';
 import * as Volume from 'resource:///org/gnome/shell/ui/status/volume.js';
 
@@ -108,12 +107,7 @@ class MprisLabel extends PanelMenu.Button {
 			new_channels.push(Math.round((channel[0] + channel[1]) / 2));
 		});
 
-		let mixedColor = new Cogl.Color({
-			red: new_channels[0],
-			green: new_channels[1],
-			blue: new_channels[2],
-			alpha: new_channels[3]
-		});
+		let mixedColor = Clutter.Color.new(new_channels[0],new_channels[1],new_channels[2],new_channels[3]);
 		let color_str = mixedColor.to_string();
 		this.unfocusColor = color_str.substring(0,7); //ignore alpha channel
 	}
--- a/src/metadata.json
+++ b/src/metadata.json
@@ -2,8 +2,8 @@
 	"uuid": "mprisLabel@moon-0xff.github.com",
 	"name": "Media Label and Controls (Mpris Label)",
 	"description": "Display a label in your panel with the song/title/album/artist information available from an mpris compatible player. You can also control the player, raise/lower its volume, customize the label, and a lot more! This extension works with Spotify, Vlc, Rhythmbox, Firefox, Chromium, and (probably) any MPRIS compatible player.",
-	"version": 37,
-	"shell-version": [ "47" ],
+	"version": 38,
+	"shell-version": [ "45", "46" ],
 	"url": "https://github.com/Moon-0xff/gnome-mpris-label",
 	"settings-schema": "org.gnome.shell.extensions.mpris-label"
 }
